python/cuml/manifold/t_sne.pyx

```diff
@@ -427,30 +430,34 @@ class TSNE(Base):
             X contains a sample per row.
             Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
             ndarray, cuda array interface compliant array like CuPy
+        convert_dtype : bool, optional (default = True)
+            When set to True, the fit_transform method will automatically
+            convert the inputs to np.float32.
 
         Returns
         --------
         X_new : array, shape (n_samples, n_components)
                 Embedding of the training data in low-dimensional space.
         """
-        self.fit(X)
+        self.fit(X, convert_dtype)
 
         if isinstance(X, cudf.DataFrame):
-            if isinstance(self.Y, cudf.DataFrame):
-                return self.Y
+            if isinstance(self.embedding_, cudf.DataFrame):
+                return self.embedding_
             else:
-                return cudf.DataFrame.from_gpu_matrix(self.Y)
+                return cudf.DataFrame.from_gpu_matrix(self.embedding_)
         elif isinstance(X, np.ndarray):
-            data = self.Y.copy_to_host()
-            del self.Y
+            data = self.embedding_.copy_to_host()
```

Comment:
I'm really looking forward to Dante's new Array changes making this all more automated!